clang/test/CodeGenSYCL/accessor_no_alias_property.cpp

```diff
@@ -0,0 +1,36 @@
+// RUN: %clang_cc1 -fsycl-is-device -triple spir64-unknown-unknown-sycldevice -emit-llvm %s -o - | FileCheck %s
+// check that noalias parameter attribute is emitted when no_alias accessor property is used
+// CHECK: define {{.*}}spir_kernel void @_ZTSZ4mainE16kernel_function1({{.*}} noalias {{.*}} %_arg_, {{.*}})
+
+// check that noalias parameter attribute is NOT emitted when it is not used
+// CHECK: define {{.*}}spir_kernel void @_ZTSZ4mainE16kernel_function2{{.*}} !kernel_arg_buffer_location
+// CHECK-NOT: define {{.*}}spir_kernel void @_ZTSZ4mainE16kernel_function2({{.*}} noalias {{.*}}
```

Comment:
Lines 6 and 7 here are broken.  The first checks to see if that kernel is defined, and the 2nd just checks to see if it is defined AGAIN.  This is not correct.  Presumably there needs to be a 'CHECK-NOT'/'CHECK-SAME' combo, like in the example here: 
https://llvm.org/docs/CommandGuide/FileCheck.html#the-check-same-directive

Reply:
PR - https://github.com/intel/llvm/pull/4283